src/Workspaces/Core/Portable/CodeGeneration/Symbols/CodeGenerationSymbol.cs

```diff
@@ -228,7 +228,7 @@ public bool Equals(ISymbol other)
 
         public bool Equals(ISymbol other, SymbolEqualityComparer equalityComparer)
         {
-            return equalityComparer.Equals(this, other);
+            return this.Equals(other);
```

Comment:
>this [](start = 19, length = 4)

The implications of this change aren't clear to me.  I suspect this is all the old implementation did at the end of the call chain, but removing the call chain may make the compiler more fragile to changes in that chain (that should have affected this).

Reply:
> The implications of this change aren't clear to me.  I suspect this is all the old implementation did at the end of the call chain, but removing the call chain may make the compiler more fragile to changes in that chain (that should have affected this).

I changed the way we deal with equality. It used to be that SymbolEqualityComparer was responsible for implementing equality, hence old code was delegating to it. Now, each object is responsible for providing implementation, SymbolEqualityComparer can be used to figure out what comparison to perform. The SymbolEqualityComparer.Equals is changed to call ```ISymbol.Equals(ISymbol other, SymbolEqualityComparer equalityComparer)``` (this method). The implementation of this method simply preserves original behavior, i.e. the new implementation is equivalent to what SymbolEqualityComparer.Equals used to do for this type. 

---
In reply to: [339834375](https://github.com/dotnet/roslyn/pull/39498#discussion_r339834375) [](ancestors = 339834375)